### ISSUE TITLE
[Streaming] BotFrameworkAdapter.useWebSockets() should be aware of the messaging endpoint path

### ISSUE THREAD

**Issue by stevengum** [microsoft/botbuilder-js#1387]
**Is your feature request related to a problem? Please describe.**
Bots using WebSockets should only begin the process of authenticating Upgrade requests when the request hits the registered endpoint in the bot's configuration on Azure.

Currently, users would have to inspect the `req.path` or `req.url` to determine if they should attempt to complete the Upgrade.

**Example:**

![image](https://user-images.githubusercontent.com/14935595/68232168-5ad95e80-ffb1-11e9-9ad6-da27c0af906d.png)

___

**Describe the solution you'd like**
Explore adding a protected "endpoint path" property to the BotFrameworkAdapter. If no property is passed in, the bot should use the "normal" path of `/api/messages`.

```ts
export interface BotFrameworkAdapterSettings {
  /*
   * 
   */
  messagesPath: string;
}

export class BotFrameworkAdapter {
   public static apiMessagesPath: string = '/api/messages';
   // ...

   constructor(settings?: BotFrameworkAdapterSettings) {
        if (!this.settings.messagesPath) {
            this.settings.messagesPath = BotFrameworkAdapter.apiMessagesPath;
        }
   }
}
```

And the ability to pass it in to `useWebSockets()` (after #1386):

```ts
public async useWebSockets(
    req: http.IncomingMessage,
    socket: net.Socket,
    head: Buffer,
    logic: (context: TurnContext) => Promise<void>,
    messagesPath?: string): Promise<void> {

    messagesPath = typeof(messagesPath) === 'string' ? messagesPath : this.settings.messagesPath; 
    if (req.method === GET && req.path === messagesPath) {
        // If the request is a proper WebSocket upgrade request and hit the registered bot
        // endpoint, complete the upgrade
    }
    // Else do nothing
}
```
___

**Describe alternatives you've considered**
Having the developer manually inspect the request before calling `BotFrameworkAdapter.useWebSockets()`

[enhancement]

**Comment by stevengum** [comment]
This, _enhancement_, is somewhat odious as the BotFrameworkAdapter is now also aware of the path it should be listening to. I'm not entirely convinced that this is the right step to take, though it may simplify developers' code.

**Comment by stevengum** [comment]
Moving this to R8 because...
1. It may not be necessary if deployed Web Apps for Bots are not expecting any other Upgrade requests.
2. It's not clear that we're doing any favors by forcing this concept on developers when they may not be expecting any other Upgrade-method requests

**Comment by stevengum** [comment]
Closing this in favor of https://github.com/microsoft/BotBuilder-Samples/issues/2196
